### PR TITLE
[#203] Add ENGINE_VERSION field to the /recommendations endpoint

### LINKED ISSUE #203: Add `ENGINE_VERSION` field to the `/recommendations` endpoint
Currently, `/recommendations` endpoint handle requests using agent version determined by `AGENT_VERSION` constant from `.env` file. The task is to add field `ENGINE_VERSION` to the `/recommendations` endpoint payload to make engine selecting process dynamic.

Notice `ENGINE_VERSION` rather than `AGENT_VERSION` - it is named in this way due to the fact that during upcoming refactoring there will be no agents at all.

Appropriate changes in recommendation requests should be implemented on the marketplace side - if needed.

On both sides tests should be added/modified.

### PR DESCRIPTION
**Pull request by Michal-Kolomanski**
Closes #203

### CODE REVIEW

File: recommender/api/endpoints/recommendations.py [hunk 1]
```diff
@@ -19,12 +18,21 @@
 api = Namespace("recommendations", "Endpoint used for getting recommendations")
 
 
-def load_agent():
-    agent_version = dotenv_values(find_dotenv()).get("AGENT_VERSION")
-    agents = {PRE_AGENT: PreAgent, RL_AGENT: RLAgent}
-    agent = agents.get(agent_version, PreAgent)()
+def load_engine(json_dict: dict) -> PreAgent or RLAgent:
```
Comment (JanKapala) [inline recommender/api/endpoints/recommendations.py:21]: PreAgent or RLAgent could be replaced with BaseAgent abstract type.

File: recommender/api/endpoints/recommendations.py [hunk 2]
```diff
@@ -19,12 +18,21 @@
...
+    """
+    Load the engine based on 'engine_version' parameter from a query
 
-    return agent
+    Args:
+        json_dict: a body from Marketplace's query
+
+    Returns:
+        An instance of pre_agent or rl_agent class
```
Comment (JanKapala) [inline recommender/api/endpoints/recommendations.py:29]: It could be a bit more elegant with a minor refactoring:

```
Returns:
        engine: An instance of pre_agent or rl_agent class
```

File: tests/factories/search_data.py [hunk 3]
```diff
@@ -35,7 +35,8 @@
 
 
 class SearchDataProvider(BaseProvider):
-    def search_phrase(self) -> str:
+    @staticmethod
```
Comment (JanKapala) [inline tests/factories/search_data.py:38]: What is the reason for this change? This will probably work (providers don't have tests in our set), but the convention here is to use a normal method instead of staticmethod - it's done this way in all other custom providers in the project.

Comment (Michal-Kolomanski) [reply tests/factories/search_data.py:38]: I saw that the 'self' object was not used in that method, but I'll stick to the convention.

**Review by JanKapala — APPROVED** [review]
Looks fine :) Btw it's good practise to add new commit for the changes required by each review - exactly as you did and next (after all approves) squash all commits into [\<issue number\>] \<issue name\> commit and force push before rebase and merge.
